File: src/main/java/me/jasper/jasperproject/JMinecraft/Item/Product/Weapons/End_Gateway.java
package me.jasper.jasperproject.JMinecraft.Item.Product.Weapons;

import lombok.Getter;
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Abilities.Teleport;
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Enchants.Sharpness;
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.ItemType;
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Rarity;
import me.jasper.jasperproject.JMinecraft.Player.Stats;
import me.jasper.jasperproject.JMinecraft.Item.JItem;
import me.jasper.jasperproject.JMinecraft.Item.Util.Factory;
import net.kyori.adventure.text.Component;
import org.bukkit.*;

import java.util.List;
import java.util.Map;

public class End_Gateway extends JItem implements Factory {
    @Getter private final static End_Gateway instance = new End_Gateway();
    public End_Gateway() {
        super("End Gateway", Material.GOLDEN_SHOVEL, Rarity.EPIC, ItemType.SWORD,  "END_GATEWAY");
        Map<Stats, Float> stats = this.getStats();
        stats.put(Stats.DAMAGE, 39f);
        stats.put(Stats.STRENGTH, 30f);
        stats.put(Stats.MANA, 50f);
        stats.put(Stats.SPEED, 10f);
        stats.put(Stats.ATTACK_SPEED, 5f);
        this.setVersion(getVersion());
        this.getEnchants().addAll(List.of(new Sharpness()));
        this.getAbilities().add(new Teleport(10, 0.2f));
    }

    @Override
    public JItem create() {
        return this;
    }

    @Override
    protected List<Component> createLore() {
        return List.of();
    }
}

File: src/main/java/me/jasper/jasperproject/JMinecraft/Item/Product/Weapons/EndGateway.kt
package me.jasper.jasperproject.JMinecraft.Item.Product.Weapons

import lombok.Getter
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Abilities.Teleport
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Enchants.Sharpness
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.ItemType
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Rarity
import me.jasper.jasperproject.JMinecraft.Item.JItem
import me.jasper.jasperproject.JMinecraft.Item.Util.Factory
import me.jasper.jasperproject.JMinecraft.Player.Stats
import net.kyori.adventure.text.Component
import org.bukkit.Material

class EndGateway : JItem("End Gateway", Material.GOLDEN_SHOVEL, Rarity.EPIC, ItemType.SWORD, "END_GATEWAY"),
    Factory {
    init {
        val stats = this.stats
        stats[Stats.DAMAGE] = 39f
        stats[Stats.STRENGTH] = 30f
        stats[Stats.MANA] = 50f
        stats[Stats.SPEED] = 10f
        stats[Stats.ATTACK_SPEED] = 5f
        this.version = version
        enchants.addAll(listOf(Sharpness()))
        abilities.add(Teleport(10, 0.2f))
    }

    override fun create(): JItem {
        return this
    }

    override fun createLore(): List<Component> {
        return listOf()
    }

    companion object {
        @Getter
        private val instance = EndGateway()
    }
}
